components/client-attestation-mgt/org.wso2.carbon.identity.client.attestation.mgt/src/main/java/org/wso2/carbon/identity/client/attestation/mgt/internal/ClientAttestationMgtServiceComponent.java

```diff
@@ -43,28 +55,102 @@
+        ClientAttestationMgtDataHolder.getInstance()
+                .setAppleAttestationRootCertificate(getAppleAttestationRootCertificate());
+        ClientAttestationMgtDataHolder.getInstance()
+                .setAppleAttestationRevocationCheckEnabled(loadAppleAttestationRevocationCheckEnabled());
+    }
+
+    /**
+     * Loads the status of Apple attestation revocation check from the configuration.
+     *
+     * @return True if revocation check is enabled, false otherwise.
+     */
+    private boolean loadAppleAttestationRevocationCheckEnabled() {
+
+        return Boolean.parseBoolean(IdentityUtil.getProperty(APPLE_ATTESTATION_REVOCATION_CHECK_ENABLED));
+    }
+
+    /**
+     * Retrieves the Apple attestation root certificate from the configured file path.
+     *
+     * @return The Apple attestation root certificate, or null if not found.
+     */
+    private X509Certificate getAppleAttestationRootCertificate() {
+
+        try {
+            String appleAttestationRootCertificatePath =
+                    IdentityUtil.getProperty(APPLE_ATTESTATION_ROOT_CERTIFICATE_PATH);
+
+            if (StringUtils.isNotBlank(appleAttestationRootCertificatePath)) {
+                CertificateFactory certificateFactory = CertificateFactory.getInstance("X509");
+                FileInputStream fileInputStream = new FileInputStream(appleAttestationRootCertificatePath);
+                X509Certificate appleAttestationRootCertificate =
+                        (X509Certificate) certificateFactory.generateCertificate(fileInputStream);
+
+                // Warn if the certificate is expiring soon
+                if (isCertificateExpiringSoon(appleAttestationRootCertificate)) {
+                    LOG.warn("Provided apple attestation root certificate is going to expire soon. " +
+                            "Please add the latest certificate.");
+                }
+                return appleAttestationRootCertificate;
+            } else {
+                LOG.warn("Apple attestation root certificate path is not configured.");
+            }
+        } catch (CertificateException | FileNotFoundException e) {
+            LOG.warn("Apple attestation root certificate not found.", e);
         }
+        return null;
+    }
+
+    /**
+     * Checks if the given X.509 certificate is expiring within 30 days.
+     *
+     * @param certificate The X.509 certificate to check.
+     * @return True if the certificate is expiring soon, false otherwise.
+     */
+    private boolean isCertificateExpiringSoon(X509Certificate certificate) {
+
+        Date currentDate = new Date();
+        Date expirationDate = certificate.getNotAfter();
+
+        // Calculate the difference in days
+        long differenceInDays = (expirationDate.getTime() - currentDate.getTime()) / (24 * 60 * 60 * 1000);
+
+        // Check if the certificate is expiring within a month
+        return differenceInDays <= 30;
```

Comment:
let's get these numbers to constants. And can you check what the ideal timeframe would be for a such warn log.

Reply:
checked with team, 90 days is the best practise